frontend/views/utils/captureLogs.js

```diff
@@ -0,0 +1,102 @@
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS } from '~/frontend/utils/events.js'
+
+const SIZE_LIMIT_KB = 100
+const isNewSession = !sessionStorage.getItem('NEW_SESSION')
+const initialLogs = localStorage.getItem(CAPTURED_LOGS)
+
+// Mark in localStorage that is a new visit (and session)
+if (isNewSession) { sessionStorage.setItem('NEW_SESSION', 1) }
+
+const setNewEntry = [
+  ...(JSON.parse(initialLogs) || []),
+  {
+    type: isNewSession ? 'NEW_SESSION' : 'NEW_VISIT',
+    msg: Date.now()
+  }
+]
+
+localStorage.setItem(CAPTURED_LOGS, JSON.stringify(setNewEntry))
+
+// Create a middleware to capture new console logs...
+const giConsole = (function (csl) {
+  return {
+    log: function (...args) {
+      csl.log(...args)
+    },
+    info: function (...args) {
+      csl.info(...args)
+    },
+    warn: function (...args) {
+      csl.warn(...args)
+      captureMsg('warn', ...args)
+    },
+    error: function (...args) {
+      csl.error(...args)
+      captureMsg('error', ...args)
+    },
+    debug: function (...args) {
+      csl.debug(...args)
+      captureMsg('debug', ...args)
+    }
+  }
+}(window.console))
```

Comment:
This sounds cool but I kept my approach, so it doesn't override `log` and `info`. That way we can still see the logs origin on the devtools (ex: `main.js` instead of `captureLogs.js`). 

When it is `error` or `warn` it shows the stack trace (very helpful) but on `log` and `debug` it doesn't... do you know how we can overcome this?

![image](https://user-images.githubusercontent.com/14869087/77191776-61392480-6ad3-11ea-85a5-9dc49b967bbf.png)

Reply:
I'm not understanding. Why do you want stack traces on `log` and `debug`? We don't want stack traces for that unless we explicitly log them, for example, by logging an error object directly or `e.stack`.

Also, if you want to try different things you can for example change `obj[prop](...args)` to `obj[prop].call(obj, ...args)` or maybe `obj[prop].apply(obj, args)` (check the MDN docs, I don't remember this stuff exactly)

Reply:
It's useful to have the stack trace while developing so i can see where the message comes from and click on the file. I tried all those "different things", none of them "removed" `captureLogs.js` from the stack trace.

Reply:
@sandrina-p ah ok, i see what you're saying. You're just asking how to prevent `captureLogs.js` from appearing. I'm not sure. You might want to have a look at stackoverflow for that, to see how `err.stack` can be manipulated for this... After I review the other recent changes you pushed I can maybe help with the search for a potential solution. It's an issue but not a *huge* deal...